integrations/Rockerbox/browser.js

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable prefer-destructuring */
+/* eslint-disable class-methods-use-this */
+import get from "get-value";
+import logger from "../../utils/logUtil";
+import { NAME } from "./constants";
+import { LOAD_ORIGIN } from "../ScriptLoader";
+import { getHashFromArray } from "../utils/commonUtils";
+
+class Rockerbox {
+  constructor(config) {
+    this.clientAuthId = config.clientAuthId;
+    this.name = NAME;
+    this.customDomain = config.customDomain;
+    this.enableCookieSync = config.enableCookieSync;
+    this.eventsMap = config.eventsMap || [];
+    this.useNativeSDKToSend = config.useNativeSDKToSend;
+  }
+
+  init() {
+    logger.debug("=== In init Rockerbox ===");
+    const host = this.customDomain ? this.customDomain : "getrockerbox.com";
+    const library =
+      this.customDomain && this.enableCookieSync ? "wxyz.rb" : "wxyz.v2";
+    (function (d, RB) {
+      if (!window.RB) {
+        window.RB = RB;
+        RB.queue = RB.queue || [];
+        RB.track =
+          RB.track ||
+          function () {
+            RB.queue.push(Array.prototype.slice.call(arguments));
+          };
+        RB.initialize = function (s) {
+          RB.source = s;
+        };
+        var a = d.createElement("script");
+        a.type = "text/javascript";
+        a.async = !0;
+        a.src = `https://${host}/assets/${library}.js`;
+        a.dataset.loader = LOAD_ORIGIN;
+        const f = d.getElementsByTagName("script")[0];
+        f.parentNode.insertBefore(a, f);
+      }
+    })(document, window.RB || {});
+    window.RB.disablePushState = true;
+    window.RB.queue = [];
+    window.RB.initialize(this.clientAuthId);
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded Rockerbox===");
+    return !!window.RB && !!window.RB.loaded;
+  }
+
+  isReady() {
+    logger.debug("===In isReady Rockerbox===");
+    return !!window.RB;
+  }
+
+  identify(rudderElement) {
+    logger.debug("===In Rockerbox Identify===");
+    const { message } = rudderElement;
+    const { userId } = message;
+    const email =
+      get(message, "context.traits.email") || get(message, "traits.email");
+    window.RB.track("identify", {
+      external_id: userId,
```

Comment:
What would be the situation when the user is totally anonymous?

Reply:
A totally anonymous user, without any user Id should not be created, as `userId` which is mapped to `external_id` in RockerBox, and external_id is a true first-party identifier. [Ref](https://help.rockerbox.com/docs/identify-calls#:~:text=Details-,external_id,-This%20user%20ID)

Reply:
Okay

Reply:
In that case we should put some log at-least

Reply:
Added a debug log for now.